core/exec/src/mill/exec/ExecutionContexts.scala

```diff
@@ -38,7 +38,15 @@ private object ExecutionContexts {
       threadCount0,
       0,
       TimeUnit.SECONDS,
-      new LinkedBlockingQueue[Runnable]()
+      // Use a `Deque` rather than a normal `Queue`, with the various `poll`/`take`
+      // operations reversed, providing elements in a LIFO order. This ensures that
+      // child `fork.async` tasks always take priority over parent tasks, avoiding
+      // large numbers of blocked parent tasks from piling up
+      new LinkedBlockingDeque[Runnable]() {
+        override def poll(): Runnable = super.pollLast()
+        override def poll(timeout: Long, unit: TimeUnit): Runnable = super.pollLast(timeout, unit)
+        override def take(): Runnable = super.takeLast()
```

Comment:
maybe we can use a forkjoin pool instead, which can configure the queue with fifo or lifo with `async` parameter.

Reply:
Forkjoin has weird behavior where `Await` is re-entrant, we used to use it and had to swap it out